src/batch_norm_api.cpp

```diff
@@ -88,27 +90,156 @@ miopenBatchNormalizationForwardInference(miopenHandle_t handle,
+                                               bn_mode,
+                                               alphaDataDiff,
+                                               betaDataDiff,
+                                               alphaParamDiff,
+                                               betaParamDiff,
+                                               xDesc,
+                                               x,
+                                               dyDesc,
+                                               dy,
+                                               dxDesc,
+                                               dx,
+                                               bnScaleBiasDiffDesc,
+                                               nullptr,
+                                               nullptr,
+                                               nullptr,
+                                               bnScale,
+                                               resultBnScaleDiff,
+                                               resultBnBiasDiff,
+                                               epsilon,
+                                               savedMean,
+                                               savedInvVariance);
+}
+
+extern "C" miopenStatus_t
+miopenBatchNormalizationForwardInference_V2(miopenHandle_t handle,
+                                            miopenBatchNormMode_t bn_mode,
+                                            void* alpha,
+                                            void* beta,
+                                            const miopenTensorDescriptor_t xDesc,
+                                            const void* x,
+                                            const miopenTensorDescriptor_t yDesc,
+                                            void* y,
+                                            const miopenTensorDescriptor_t scaleDesc,
+                                            const miopenTensorDescriptor_t BiasDesc,
+                                            const miopenTensorDescriptor_t estMeanDesc,
+                                            const miopenTensorDescriptor_t estVarianceDesc,
+                                            void* bnScale,
+                                            void* bnBias,
+                                            void* estimatedMean,
+                                            void* estimatedVariance,
+                                            double epsilon)
 {
     MIOPEN_LOG_FUNCTION(handle,
                         bn_mode,
                         xDesc,
                         x,
                         yDesc,
                         y,
-                        bnScaleBiasMeanVarDesc,
+                        scaleDesc,
+                        (BiasDesc == nullptr) ? scaleDesc : BiasDesc,
+                        (estMeanDesc == nullptr) ? scaleDesc : estMeanDesc,
+                        (estVarianceDesc == nullptr) ? scaleDesc : estVarianceDesc,
```

Comment:
If we moved these checks into the original API, and instead called _V2 with these set, then we could make these descriptors required for the _V2 API.

It's not a huge deal, but I think doing that would simplify the code a bit.

(If you decide to do this change, then the same also applies for training & backwards)

Reply:
good call